ShopApp/Domain/Repositories/SetupProviderRepository.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  SetupProviderRepository.swift
+//  ShopApp
+//
+//  Created by Evgeniy Antonov on 6/14/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import ShopApp_Gateway
+
+protocol SetupProviderRepository {
+    func setupProvider(callback: @escaping RepoCallback<Bool>)
```

Comment:
I think we need to use Void instead Bool in callback closure, just like we'll use it in authorization methods of API.